jupyterhub/apihandlers/users.py

```diff
@@ -209,14 +218,16 @@ def post(self, name):
             raise web.HTTPError(400, "%s's server is not running" % name)
         self.set_server_cookie(user)
         # a service can also ask for a user cookie
-        # this code prevent to raise an error
+        # this code prevents to raise an error
+        # cause service doesn't have 'other_user_cookies'
         if getattr(current, 'other_user_cookies', None) is not None:
             current.other_user_cookies.add(name)
 
 
 default_handlers = [
     (r"/api/users", UserListAPIHandler),
     (r"/api/users/([^/]+)", UserAPIHandler),
-    (r"/api/users/([^/]+)/server", UserServerAPIHandler),
+    (r"/api/users/([^/]+)/servers", UserCreateServerAPIHandler),
```

Comment:
> I understand the need for this but considering the Api is not very "consistent" ?

Sorry, I don't quite know what you are asking here.

> You would have different endpoints doing almost the same.

Yes, that's okay. We may be able to handle it with a simple redirect. We just need to be mindful of users.


We need to think carefully about the single-server case, since that's going to continue to be the vast majority of deployments. It's important that people be able to use JupyterHub with:

- never having a server with a name
- only `/user/:name` URL ever proxied to a single server
- never allowing a second server to start

The simplest case is the most important, and we need to be very careful that added features don't make it more complex to use JupyterHub in this simple mode, which should likely remain the default.